### PR TITLE
Use msbuild project info for running

### LINKED ISSUE #184: Use MSBuild to locate output folder and binaries for run scenarios
### Describe the bug
Today `tye run` hackishly finds the output of the binary by looking at the debug folder, picking the first TFM, and assmuming there's a file called `{ProjectName}.dll` in the output folder. Instead, we need to use msbuild to resolve the target assembly.

### PR DESCRIPTION
**Pull request by davidfowl**
- This fixes several subtle bugs around how we resolve output paths adhoc and uses msbuild to find various outputs.

Fixes #184

### CODE REVIEW

File: src/Microsoft.Tye.Core/ProjectServiceBuilder.cs [hunk 1]
```diff
@@ -28,9 +28,24 @@ public ProjectServiceBuilder(string name, FileInfo projectFile)
         // This is always set on the ApplicationFactory codepath.
         public string TargetFramework { get; set; } = default!;
 
+        // This is always set on the ApplicationFactory codepath.
+        public string[] TargetFrameworks { get; set; } = default!;
```
Comment (jkotalik) [inline src/Microsoft.Tye.Core/ProjectServiceBuilder.cs]: Ew, the default! isn't pretty.

File: src/Microsoft.Tye.Hosting/TransformProjectsIntoContainers.cs [hunk 2]
```diff
@@ -41,16 +41,17 @@ private async Task TransformProjectToContainer(Model.Application application, Mo
             var serviceDescription = service.Description;
             var serviceName = serviceDescription.Name;
 
-            var expandedProject = Environment.ExpandEnvironmentVariables(project.Project);
-            var fullProjectPath = Path.GetFullPath(Path.Combine(application.ContextDirectory, expandedProject));
-            service.Status.ProjectFilePath = fullProjectPath;
+            service.Status.ProjectFilePath = project.ProjectFile.FullName;
+            var targetFramework = project.TargetFramework;
```
Comment (jkotalik) [inline src/Microsoft.Tye.Hosting/TransformProjectsIntoContainers.cs]: It looks like TargetFramework can be null if TargetFrameworks isn't null. Should we assume one of the target frameworks as we are publishing on behalf of the person doing tye run?

File: src/tye/ApplicationBuilderExtensions.cs [hunk 3]
```diff
@@ -52,7 +52,17 @@ public static Tye.Hosting.Model.Application ToHostingApplication(this Applicatio
                 }
                 else if (service is ProjectServiceBuilder project)
                 {
-                    var projectInfo = new ProjectRunInfo(project.ProjectFile.FullName, project.Args, project.Build);
+                    if (project.TargetFrameworks.Length > 1)
```
Comment (jkotalik) [inline src/tye/ApplicationBuilderExtensions.cs]: Ah nvrm, I see that you are checking here as well. I think it's fine then.

File: src/Microsoft.Tye.Core/ProjectServiceBuilder.cs (hunk 1 above)
Comment (rynowak) [reply src/Microsoft.Tye.Core/ProjectServiceBuilder.cs]: get used to it. It's either massive constructor or this 😆

**Pull request merged**